packages/itwin/reports-config-widget/src/widget/components/BulkExtractor.ts

```diff
@@ -21,23 +21,30 @@ export default class BulkExtractor {
   private _extractionClientApi: ExtractionClient;
   private _accessToken: () => Promise<string>;
   private _reportStates = new Map<string, ExtractionStates>();
+  private _iModelStates = new Map<string, ExtractionStates>();
   private _timeFetched = new Date();
-  private _reportIds: string[];
-
-  constructor(apiConfig: ReportsApiConfig, reportIds: string[]) {
+  private _iModelRun = new Map<string, string>();
+  private _iModelToast = new Map<string, boolean>();
```

Comment:
I think `_iModelToast` could be changed to a `Set<string>`. You could also rename this to `_toastediModels` or something like that, because it keeps track of which iModels already appeared in a toast message.

Reply:
`_iModelToast` changed to `Set<string>`